flink-staging/flink-streaming/flink-streaming-core/src/main/java/org/apache/flink/streaming/api/functions/source/MessageAcknowledingSourceBase.java

```diff
@@ -144,6 +152,20 @@ protected void addId(Id id) {
 	
 	@Override
 	public SerializedCheckpointData[] snapshotState(long checkpointId, long checkpointTimestamp) throws Exception {
+		if (idsForCurrentCheckpoint == null) {
+			LOG.debug("snapshotState() requested on not yet opened source; returning null.");
+			return null;
+		}
+		if (!running) {
+			LOG.debug("snapshotState() called on closed source");
+			return null;
+		}
+
+		if (LOG.isDebugEnabled()) {
+			LOG.debug("Snapshotting state. Messages: {}, checkpoint id: {}, timestamp: {}",
```

Comment:
The list of IDs can be very long. Not sure if it is a good idea to put that into a log statement (single line, string)